### PR TITLE
Iso format

### PR DESCRIPTION
**Pull request by juanlang**

### CODE REVIEW

File: u2f-tests/HID/u2f_util.cc [hunk 1]
```diff
@@ -385,6 +371,40 @@ int U2Fob_apdu(struct U2Fob* device,
...
+  buf[3] = P2;
+
+  uint8_t offs = 4;
+  if (nc) {
+    buf[offs++] = 0;  // extended length
+    buf[offs++] = (out.size() >> 8) & 255;
+    buf[offs++] = (out.size() & 255);
+    memcpy(buf + offs, out.data(), out.size());
+    offs += out.size();
+  }
```
Comment (mschilder123) [inline u2f-tests/HID/u2f_util.cc]: else?
Or rather,
buf[4] = 0;
offs = 5;
if (nc) {
}

(and no !nc section)

File: u2f-tests/HID/u2f_util.cc [hunk 2]
```diff
@@ -385,6 +371,40 @@ int U2Fob_apdu(struct U2Fob* device,
...
+  if (!nc) {
+    // When there are no data sent, an extra 0 is necessary prior to Le.
+    buf[offs++] = 0;
+  }
+  buf[offs++] = 0;
+  buf[offs] = 0;
+
+  return U2Fob_exchange(device, buf, bufSize, in);
```
Comment (mschilder123) [inline u2f-tests/HID/u2f_util.cc]: instead of computing bufSize in separate fashion, use offs here?

File: u2f-tests/HID/u2f_util.cc (hunk 1 above)
Comment (juanlang) [inline u2f-tests/HID/u2f_util.cc]: This was a deliberate choice, to encode Lc and Le separately, even if the branches look a little funny. I added comments that I hope will clarify why it's structured this way.

File: u2f-tests/HID/u2f_util.cc (hunk 2 above)
Comment (juanlang) [inline u2f-tests/HID/u2f_util.cc]: Word. Done.

File: u2f-tests/HID/u2f_util.h [hunk 3]
```diff
@@ -112,6 +112,14 @@ int U2Fob_recv(struct U2Fob* device, uint8_t* cmd,
 // returns
 //   negative error
 //   positive sw12, e.g. 0x9000, 0x6985 etc.
+int U2Fob_exchange(struct U2Fob* device,
```
Comment (leshi) [inline u2f-tests/HID/u2f_util.h]: Slightly confused by choice of function name and comment.  Would be nice if something about the comment and method name made it clear how this is different from ```U2Fob_apdu()```.

**Pull request merged**